lib/preprocessor.cpp

```diff
@@ -338,6 +344,8 @@ std::list<Directive> Preprocessor::createDirectives(const simplecpp::TokenList &
                     directive.str += "include";
                 else
                     directive.str += tok2->str();
+                if (!tok2->next || tok2->next->location.line != directive.linenr)
+                    directive.endToken = tok2;
```

Comment:
If there is a comment after the directive then the endToken is not set. It would be better to set endToken unconditionally.